src/rust/aggregator/src/data/leaderboards.rs

```diff
@@ -0,0 +1,261 @@
+use anyhow::anyhow;
+use bigdecimal::BigDecimal;
+use chrono::{DateTime, Duration, Utc};
+use sqlx::{PgConnection, PgPool, Postgres, Transaction};
+
+use super::{Data, DataAggregationError, DataAggregationResult};
+
+pub struct Leaderboards {
+    pool: PgPool,
+    last_indexed_timestamp: Option<DateTime<Utc>>,
+}
+
+impl Leaderboards {
+    pub fn new(pool: PgPool) -> Self {
+        Self {
+            pool,
+            last_indexed_timestamp: None,
+        }
+    }
+}
+
+struct Competition {
+    id: i32,
+    start: DateTime<Utc>,
+    end: DateTime<Utc>,
+    prize: i64,
+    market_id: BigDecimal,
+    frontends_required: Vec<String>,
+}
+
+#[async_trait::async_trait]
+impl Data for Leaderboards {
+    fn ready(&self) -> bool {
+        self.last_indexed_timestamp.is_none()
+            || self.last_indexed_timestamp.unwrap() + Duration::seconds(5) < Utc::now()
+    }
+
+    async fn process_and_save_historical_data(&mut self) -> DataAggregationResult {
+        self.process_and_save_internal().await
+    }
+
+    fn poll_interval(&self) -> Option<std::time::Duration> {
+        Some(std::time::Duration::from_secs(5))
```

Comment:
Let's define this with a constant rather than a magic number inline

I also think it should be something like 1.5 seconds, which may require updating other aggregator logic

The idea being that the leaderboard frontend can poll the backend every 2 seconds or so?